components/soyosource_virtual_meter/soyosource_virtual_meter.cpp

```diff
@@ -20,6 +20,11 @@ void SoyosourceVirtualMeter::setup() {
              this->last_power_demand_);
 
     this->last_power_demand_received_ = millis();
+
+    //if update intervall is set to never, trigger the update() method from here
+    if (this->get_update_interval() == -1) {
```

Comment:
```suggestion
    if (this->get_update_interval() == SCHEDULER_DONT_RUN) {
```

See esphome/core/component.h

```
static const uint32_t SCHEDULER_DONT_RUN = 4294967295UL;
```

Reply:
yeah new c++ stuff for me. 😁 (did not code in c++ before this project.